### PR TITLE
Add Flake8 comprehensions to pre-commit

### PR DESCRIPTION
**Pull request by CaedenPH**
### Describe your change:
Implements (#7233)

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by realstealthninja — APPROVED** [review]

File: hashes/enigma_machine.py [hunk 1]
```diff
@@ -1,8 +1,8 @@
 alphabets = [chr(i) for i in range(32, 126)]
-gear_one = [i for i in range(len(alphabets))]
-gear_two = [i for i in range(len(alphabets))]
-gear_three = [i for i in range(len(alphabets))]
-reflector = [i for i in reversed(range(len(alphabets)))]
+gear_one = range(len(alphabets))
+gear_two = range(len(alphabets))
+gear_three = range(len(alphabets))
+reflector = reversed(range(len(alphabets)))
```
Comment (dhruvmanila) [inline hashes/enigma_machine.py:5]: The old code is constructing a list, but the updated code is giving us a `range` object. Is this ok?

**Review by dhruvmanila — APPROVED** [review]
One small doubt, otherwise LGTM! 🎉

File: hashes/enigma_machine.py (hunk 1 above)
Comment (realstealthninja) [reply hashes/enigma_machine.py:5]: no because range returns a list

Comment (CaedenPH) [reply hashes/enigma_machine.py:5]: Range returns a `range` object, which does not have the same methods as a list. Now that I realise, yes, I believe this is a problem. I will encase these range objects in the list function

**Pull request merged**